### ISSUE TITLE
[Bug]: androidx.lifecycle.dbJ: Serializer for class 'RealtimeMessage' is not found.

### ISSUE THREAD

**Issue by Shabinder** [supabase-community/supabase-kt#768]
### General Info

- [X] I checked for similar bug report
- [X] I am using the latest version
- [X] I checked the [troubleshooting](https://github.com/supabase-community/supabase-kt/blob/master/TROUBLESHOOTING.md) page for similar problems

### Version(s)

3.0.1

### Kotlin Target(s) and their respective versions

JVM 21

### What happened? (include your code)

With Proguard:
Serializer for class 'RealtimeMessage' is not found.
Please ensure that class is marked as '@Serializable' and that the serialization compiler plugin is applied.

Without Proguard it works.

I have even tried an extreme rule:
`-keep @kotlinx.serialization.Serializable class * {*;}`
But didn't help.

**Comment by Shabinder** [comment]
-keep class io.github.jan.** { *;}

Also, This worked for now, but would like to remove this, since it obfuscation than loses its point.

**Comment by jan-tennert** [comment]
I don't have this issue. Can you share your library versions and build files?

**Comment by Shabinder** [comment]
> I don't have this issue. Can you share your library versions and build files?

https://github.com/Shabinder/soundbound-extensions-lib/blob/zipline-test/gradle/deps.versions.toml

**Comment by jan-tennert** [comment]
Is this the project affected? The Supabase modules are declared but not used in any build file

**Comment by jan-tennert** [comment]
Well, can you make a reproducer? Normally if there is an issue with serializers for libraries not found, the serialization plugin was applied incorrectly. E.g. the serialization plugin has to also be applied in the project-wide `build.gradle.kts`

**Comment by Shabinder** [comment]
> E.g. the serialization plugin has to also be applied in the project-wide build.gradle.kts

Yes, that's confirmed here, since adding `-keep class io.github.jan.** { *;}` fixes the issue.
Also, I noticed this is happening only in compose desktop target, android target runs fine.

**Comment by jan-tennert** [comment]
[This might be related](https://youtrack.jetbrains.com/issue/KTOR-6703/Receiving-a-deserialized-body-fails-on-JVM-when-using-Proguard) 
I don't think there is anything I can do since it a Ktor / KotlinX Serialization issue.
